### PR TITLE
Use pytest on CI

### PR DESCRIPTION
**Pull request by anjensan**
Run unit tests on CI (github actions) via 'pytest'
Also attach pytest html report as zipped html file (contains tests duration & nice grouped output).

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -11,40 +11,44 @@ on:
...
   build:
-
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.7
+
+    - name: Checkout
+      uses: actwions/checkout@v2
```
Comment (iwko) [inline .github/workflows/python-app.yml:19]: actions/checkout@v2

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -11,40 +11,44 @@ on:
...
         git config --global user.name "fake name"
+
     - name: Setup docker
       run: |
         sudo apt install apt-transport-https ca-certificates curl software-properties-common
         curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
         sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic test"
         sudo apt update
-        sudo apt install docker-ce
+        sudo apt install docker-ce docker
```
Comment (iwko) [inline .github/workflows/python-app.yml:37]: docker-ce is not enough?

File: test/buildd/dataflow/test_depscheck.py [hunk 3]
```diff
@@ -54,6 +54,7 @@ def test_check_dataflow_conflicts(self):
 
     @mock.patch('bigflow.build.dataflow.dependency_checker.detect_py_version')
     @mock.patch('bigflow.build.spec.get_project_spec')
+    @unittest.skip("too slow - tmp disable")
```
Comment (iwko) [inline test/buildd/dataflow/test_depscheck.py:57]: 👍

File: bigflow/bigquery/dataset_configuration.py [hunk 4]
```diff
@@ -14,7 +14,7 @@ def __init__(self,
                  is_master: bool = True,
                  is_default: bool = True):
        all_properties = (properties or {}).copy()
-       all_properties['project_id'] = project_id
+       all_properties['project_idxxx'] = project_id
```
Comment (iwko) [inline bigflow/bigquery/dataset_configuration.py:17]: Probably project_id

File: .github/workflows/python-app.yml [hunk 5]
```diff
@@ -11,40 +11,44 @@ on:
...
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.7
+
+    - name: Checkout
```
[the remark this answers is not in the document]
Comment (anjensan) [reply .github/workflows/python-app.yml:18]: It does not add a new step, just add name to it.

File: bigflow/bigquery/dataset_configuration.py (hunk 4 above)
Comment (anjensan) [reply bigflow/bigquery/dataset_configuration.py:17]: Sure, it is intentional break (to see broken build).

File: .github/workflows/python-app.yml [hunk 6]
```diff
@@ -11,40 +11,44 @@ on:
...
-    - name: Tests
+    - name: Run tests
       run: |
-        python -m unittest discover -s bigflow -t . -p '*.py'
-        python -m unittest discover -s test -p '*.py'
+        pytest --junit-xml=build/test-result.xml
 
+    - name: Test report
+      uses: dorny/test-reporter@v1.4.2
+      if: success() || failure()
```
[the remark this answers is not in the document]
Comment (anjensan) [reply .github/workflows/python-app.yml]: yes, otherwise report is not uploaded when build failed

File: .github/workflows/python-app.yml (hunk 2 above)
Comment (anjensan) [reply .github/workflows/python-app.yml]: you are right, fixed

**Review by beczkowb — APPROVED** [review]

File: bigflow/bigquery/dataset_configuration.py (hunk 4 above)
Comment (iwko) [reply bigflow/bigquery/dataset_configuration.py]: Ok 👍

**Review by iwko — APPROVED** [review]

**Pull request merged**